### DISCUSSION TITLE
Additonally export innernet-server as a lib

### DISCUSSION THREAD

**Discussion by sqrtsanta** [tonarino/innernet#319] [Ideas]
Hi, folks!

Would you be down to pulling certain parts from `innernet/server/main.rs` to `innernet/server/lib.rs`? This would enable users to build own wrappers of innernet-server in case they need something more than CLI:

- http/gprc server for innernet-server commands like add-peer/add-cidr/etc
- send toml invitation right away without checking file-system
- etc

I would be glad to make PR, just wanted to make sure you are ok with possible changes

**Comment by strohel** [comment]
Hi @sqrtsanta, this sounds fine, as long as the boundary between bin and lib stays (subjectively) reasonable. I'll be happy to review PRs.

**Comment by sqrtsanta** [comment]
Gotcha, thanks!
